core/src/robot/BasicTransmissions.cc

```diff
@@ -0,0 +1,66 @@
+#include <algorithm>
+
+#include "jiminy/core/utilities/Helpers.h"
+
+#include "jiminy/core/robot/BasicTransmissions.h"
+
+
+namespace jiminy
+{
+    SimpleTransmission::SimpleTransmission(std::string const & name) :
+    AbstractTransmissionBase(name)
+    {
+        std::cout << "construct simple transmission" << std::endl;
+    }
+
+    hresult_t SimpleTransmission::initialize(std::vector<std::string> const & jointNames,
+                                             std::vector<std::string> const & motorNames)
+    {
+        if (jointNames.size() != 1 || motorNames.size() != 1)
+        {
+            PRINT_ERROR("A simple transmission only supports 1-to-1 mappings of motors and joints.");
```

Comment:
print error is not enough, you also need to `return hresult_t::ERROR_BAD_INPUT;`